extension/app/ts/components/simulationExplaining/identifyTransaction.tsx

```diff
@@ -3,55 +3,82 @@ import { CHAINS, FourByteExplanations, isSupportedChain, MAKE_YOU_RICH_TRANSACTI
 import { SimulatedAndVisualizedTransaction } from '../../utils/visualizer-types.js'
 import { getSwapName, identifySwap } from './SwapTransactions.js'
 
-type TRANSACTION_TYPE = 'MakeYouRichTransaction' | 'NormalTransaction'
+type TRANSACTION_TYPE = 'MakeYouRichTransaction' | 'ArbitaryContractExecution' | 'EtherTransfer' | 'Swap' | 'ContractFallbackMethod'
 
-export function nameTransaction(transaction: SimulatedAndVisualizedTransaction, activeAddress: bigint ) {
-	if (identifyTransaction(transaction, activeAddress) === 'MakeYouRichTransaction') {
-		return 'Simply making you rich'
-	}
-	if (transaction.input.length == 0) return 'Ether Transfer'
-
-	//TODO: add simple token transfer
-
-	const identifiedSwap = identifySwap(transaction)
-	if (identifiedSwap) return getSwapName(identifiedSwap, transaction.chainId)
-
-	const fourByte = get4Byte(transaction.input)
-	if (fourByte === undefined) return 'Contract Fallback Method'
-	const explanation = FourByteExplanations.get(fourByte)
-	return explanation !== undefined ? explanation : 'Contract Execution'
+type IdenttifiedTransaction = {
+	type: TRANSACTION_TYPE,
+	title: string,
+	signingAction: string,
+	simulationAction: string,
+	rejectAction: string,
 }
 
+export function identifyTransaction(transaction: SimulatedAndVisualizedTransaction, activeAddress: bigint): IdenttifiedTransaction {
+	const chainString = transaction.chainId.toString()
 
-export function nameTransactionAction(transaction: SimulatedAndVisualizedTransaction, activeAddress: bigint ) {
-	if (identifyTransaction(transaction, activeAddress) === 'MakeYouRichTransaction') {
-		return 'Rich'
+	if (isSupportedChain(chainString)
+		&& CHAINS[chainString].eth_donator === transaction.from.address
+		&& transaction.to?.address === activeAddress
+		&& transaction.type === MAKE_YOU_RICH_TRANSACTION.type
+		&& transaction.maxFeePerGas === MAKE_YOU_RICH_TRANSACTION.maxFeePerGas
+		&& transaction.maxPriorityFeePerGas === MAKE_YOU_RICH_TRANSACTION.maxPriorityFeePerGas
+		&& transaction.input.toString() === MAKE_YOU_RICH_TRANSACTION.input.toString()
+		&& transaction.value === MAKE_YOU_RICH_TRANSACTION.value
```

Comment:
Similar to comment on previous PR, I would much rather see an envelope that contained the transaction plus some metadata about it, rather than trying to infer the transaction type based on the transaction details.

That being said, we will have to do this inference at some point for many transaction types (e.g., Swaps), so doing it for all of them isn't terrible.  Even in that case though, I feel like we should have an envelope that stores that information so we don't have to infer multiple times.

```ts
const envelope = { transaction, website, type }
```

Reply:
We are doing that inference for swaps, ether sends and token sends. That envelope would only be used to separate MAKE_YOU_RICH_TRANSACTION ? Its the only transaction that the Interceptor creates.

Reply:
My thinking was to classify transactions once as soon as the Interceptor receives them, and then store that classification in the transaction envelope's metadata.  This way everywhere we want to know the type, we already have it.  Essentially, classification happens as part of the initial deserialization/creation of the transaction.

Reply:
yeah, I do classification many times with the same function as its not that complex operation

Reply:
hmm, doing classification "when we get the transaction" won't do it, as it can change. We need to do it every time we run simulation again. Unless we make the categorization not dependent on the simulation, which would be safer, but more difficult

Reply:
The classification of a transaction changing between simulations *feels* like an edge case that we shouldn't worry about/deal with.  Do you have some thoughts on a reasonable way that would happen?

If not, then we can do classification once when we first see the transaction and then store the classification in the envelope and never do it again.

Reply:
the only reasonable way is when the transaction fails. You don't get simulation results so you cannot classify it anymore. This happens quite often with uniswap swaps timeoutting. But then the initial classification would be correct. Its a swap that now fails